runtime/modules/java/module/src/main/java/io/atlasmap/java/module/JavaValidationService.java

```diff
@@ -106,14 +101,15 @@ protected String getModuleFieldName(JavaField field) {
         return field.getName() != null ? field.getName() : field.getPath();
     }
 
-    protected void validateSourceAndTargetTypes(String mappingId, Field inputField, Field outField, List<Validation> validations) {
+    protected void validateSourceAndTargetTypes(String mappingId, Field inputField, Field outField,
+            List<Validation> validations) {
         if ((inputField instanceof JavaField && outField instanceof JavaField)
                 && ((inputField.getFieldType() == null || outField.getFieldType() == null)
-                    || (inputField.getFieldType().compareTo(FieldType.COMPLEX) == 0
-                    || outField.getFieldType().compareTo(FieldType.COMPLEX) == 0))) {
+                        || (inputField.getFieldType().compareTo(FieldType.COMPLEX) == 0
```

Comment:
I know it's not your code, but doesn't it look weird to use `compareTo()` here? I would use just `==` for the Enum